bundles/org.openhab.binding.homewizard/src/main/java/org/openhab/binding/homewizard/internal/HomeWizardHandler.java

```diff
@@ -0,0 +1,274 @@
+/**
+ * Copyright (c) 2010-2021 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.homewizard.internal;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.util.Map;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.homewizard.data.P1Payload;
+import org.openhab.core.io.net.http.HttpUtil;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.library.unit.Units;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+/**
+ * The {@link HomeWizardHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Daniël van Os - Initial contribution
+ */
+@NonNullByDefault
+public class HomeWizardHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(HomeWizardHandler.class);
+
+    private HomeWizardConfiguration config = new HomeWizardConfiguration();
+
+    private final Lock pollingJobLock = new ReentrantLock();
+    private @Nullable ScheduledFuture<?> pollingJob;
+    private int currentRefreshDelay = 0;
+
+    private final GsonBuilder builder = new GsonBuilder();
+    private final Gson gson = builder.create();
+
+    private final Lock configurationLock = new ReentrantLock(); // to protected fields accessed in init/dispose AND the
+                                                                // poller
+    private String apiURL = "";
+
+    private String meterModel = "";
+    private int meterVersion = 0;
+
+    /**
+     * Constructor
+     *
+     * @param thing The thing to handle
+     */
+    public HomeWizardHandler(Thing thing) {
+        super(thing);
+    }
+
+    /**
+     * Not listening to any commands.
+     */
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+    }
+
+    /**
+     * If a host has been specified start polling it
+     */
+    @Override
+    public void initialize() {
+        config = getConfigAs(HomeWizardConfiguration.class);
+
+        if (configure()) {
+            stopPolling(true);

```

Comment:
Removed `stopPolling` here. before initialize is called the framework will call dispose. So it's already stopped.